### PR TITLE
Api controller

### LINKED ISSUE #44: Rest-API
The API which will be used to test our program. (This will be split into multiple tasks)

### PR DESCRIPTION
**Pull request by Hjaltesorgenfrei**
Api controller to the spec. passes all tests in the Api spec folder. 
fixes #44

### CODE REVIEW

File: MiniTwit.Models/UserRepository.cs [hunk 1]
```diff
@@ -78,5 +82,78 @@ private async Task<bool> UserExists(int userId, string emailAddress)
         {
             return await _context.Users.AnyAsync(u => u.Id != userId && u.Email == emailAddress);
         }
+
+        public async Task AddFollowerAsync(int followerId, int followeeId)
+        {
+            if (followeeId == followerId)
+            {
+                //TODO: Log this to the file, but no logger is sent into the repo, so its a bit hard.
```
Comment (jlndk) [inline MiniTwit.Models/UserRepository.cs]: Can this be solved with dependency injection?

File: MiniTwit.Entities/Follows.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace MiniTwit.Entities
+{
+    public class Follows
```
Comment (jlndk) [inline MiniTwit.Entities/Follows.cs]: Isn't the convension to name Entities/Models in singular?

File: MiniTwit.Web.App/Controllers/ApiController.cs [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+            }
+
+            return (true,
+                StatusCode(403, Json(new {status = 403, error_msg = "You are not authorized to use this resource!"})));
+        }
+
+        [Route("[controller]/latest/")]
+        [HttpGet]
+        [AllowAnonymous]
+        public async Task<IActionResult> GetLatest()
```
Comment (jlndk) [inline MiniTwit.Web.App/Controllers/ApiController.cs]: Warning fra CI

File: MiniTwit.Web.App/Controllers/ApiController.cs [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+        {
+            UpdateLatest(latestMessage);
+            UpdateLatest(followPost.Latest);
+            if (followPost?.Follow != null)
+            {
+                var follower = await _userRepository.ReadAsyncByUsername(username);
+                var followee = await _userRepository.ReadAsyncByUsername(followPost.Follow);
+                if (followee == null)
+                {
+                    // TODO: This has to be another error, likely 500???
```
Comment (jlndk) [inline MiniTwit.Web.App/Controllers/ApiController.cs:201]: Would like to have these TODO's solved

**Review by jlndk — CHANGES_REQUESTED** [review]
Besides the minor points I've found, this PR should wait until #67 is merged.

File: MiniTwit.Web.App/Controllers/ApiController.cs (hunk 4 above)
Comment (Hjaltesorgenfrei) [reply MiniTwit.Web.App/Controllers/ApiController.cs:201]: Nope its supposed to be there.

**Review by Duckapple — APPROVED** [review]
Seems cool enough, and I trust Hjalte

**Review by jlndk — APPROVED** [review]
I'll trust you guys

**Pull request merged**